modules/lang-painless/src/main/java/org/elasticsearch/painless/lookup/PainlessLookupUtility.java

```diff
@@ -0,0 +1,280 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.painless.lookup;
+
+import org.objectweb.asm.Type;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Map;
+
+public final class PainlessLookupUtility {
+
+    // The following terminology is used for variable names throughout the lookup package:
```

Comment:
Please move this to a javadoc on the class, and add a general description as to why this class exists.

Reply:
Will do.